src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,429 @@
+      new KeyValue<>("bob", 888.88),
+      new KeyValue<>("alice", 999.99)
+    );
+
+    final List<KeyValue<String, Long>> inputTableRecords = Arrays.asList(
+      new KeyValue<>("alice", 1L),
+      new KeyValue<>("alice", 2L)
+    );
+
+    final List<KeyValue<String, Pair<Double, Long>>> expectedRecords = Arrays.asList(
+      // The join output will have 2L (not 1L) for the table side.  That's because, by default, the DSL enables record
+      // caching for tables, which will cause the ValueTransformerWithKey for the KTable to not observe the 1L value.
+      //
+      // If your use case needs to produce a join output with value 1L, there are several options available.
+      //
+      // 1) Set `StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG` to `0`.  This will disable record caching,
+      //    and make the table's ValueTransformerWithKey to see the 1L value.  The downside of this approach is that,
+      //    by definition, record caching is disabled for the complete topology, resulting in a larger volume of
+      //    "intermediate" updates.
+      // 2) Don't use a KTable and a ValueTransformerWithKeySupplier for managing the table-side triggering for the
+      //    join.  Instead, read the table's topic into a KStream, and then use a normal Transformer with code very
+      //    similar to what's implement in StreamTableJoinStreamWaitsForTable.  You must also create a second
+      //    state store, managed by this new table-side Transformer, to manage the table-side store manually (because
+      //    you use a KStream instead of a KTable for the table's data).  Now you have full control over when and how to
+      //    perform table-side join triggering.
+      //
+      // Note: Kafka Streams' current stream-table join semantics dictate that only a single join output will ever be
+      //       produced for a newly arriving stream-side record.  Table-side triggering of the join should only be used
+      //       to ensure (rather: to increase the chance) that, when a join output is actually produced, it contains
+      //       data from both the stream and the table.  However, table-side triggering should NOT be used to sent
+      //       multiple join outputs for the same stream-side record.
+      //
+      //       Example of wrong table-side join triggering:
+      //
+      //          Time | Stream              Table         | Join output
+      //          -----+-----------------------------------+------------------------
+      //          10   | ("alice", 999.99)                 |
+      //          20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
+      //          30   |                     ("alice", 2L) | ("alice", (999.99, 2L))
+      //
+      //       In the wrong example above, only one join output must be produced, not two.  It's up to you to decide
+      //       which one, however.
+      //
+      new KeyValue<>("alice", new Pair<>(999.99, 2L)),
```

Comment:
Because your join is built on wall-clock time, there is no guarantee that the join happens on the second table record (even if I believe that the probability that this happens is zero...). Just saying.